tests/src/Traits/MockSchemaTrait.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Drupal\Tests\graphql\Traits;
+
+use Drupal\Core\DependencyInjection\ContainerBuilder;
+use Drupal\graphql\Plugin\GraphQL\SchemaPluginManager;
+use Drupal\graphql\Plugin\GraphQL\Schemas\SchemaPluginBase;
+use Drupal\KernelTests\KernelTestBase;
+use Prophecy\Argument;
+
+trait MockSchemaTrait {
+
+  abstract function getSchemaDefinitions();
+
+  /**
+   * @var \Prophecy\Prophecy\ObjectProphecy
+   */
+  protected $schemaManagerProphecy;
+
+  public function schemaManagerFactory() {
+    return $this->schemaManagerProphecy->reveal();
+  }
+
+  protected function registerSchemaPluginManager(ContainerBuilder $container) {
+    if ($this instanceof KernelTestBase) {
```

Comment:
Why do we need this check?

Reply:
Because somebody might attach this to something else than a KernelTest. And PHPStorm gives me autocompletion this way 😛

Reply:
Right but then just use the /** typehint (via comments)?